include/pybind11/cast.h

```diff
@@ -1629,7 +1629,25 @@ struct pyobject_caster {
     template <typename T = type, enable_if_t<std::is_same<T, handle>::value, int> = 0>
     bool load(handle src, bool /* convert */) { value = src; return static_cast<bool>(value); }
 
+#ifdef PYBIND11_DISABLE_IMPLICIT_STR_FROM_BYTES
```

Comment:
This is quite a horribly long name.

But actually, it's not related to this PR, right? Introducing it seems like it could be another discussion/PR, since maybe we ought to deprecate this `str`-from-`bytes` behavior so it can be removed in a pybind11 3.0.0?

Reply:
It's not meant to be a public or permanent macro. I could also just put a couple comments marking the begin and end of the part that implements the implicit decoding. The macro has the advantage (IMO) that it stands out more and that we can slightly more easily experiment in the run-up for version 3.0.0 (you can just put a `#define` right before). I don't feel very strongly about macro versus comment; only that we should have some demarkation.